.github/workflows/main.yml

```diff
@@ -47,4 +47,5 @@ jobs:
 
       # If the version has been incremented (with `yarn changeset version`), publish package
       - name: Publish on version increment
+        if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main' }}
```

Comment:
the ref check for main is to future proof this line if we expand the pr branch matching  on line 11/12/13

Reply:
I think I'd prefer to keep the build separate

Reply:
I think I'd prefer to keep the build separate